### ISSUE TITLE
Unify time and size units

### ISSUE THREAD

**Issue by petrcermak** [google/trace-viewer#1055]
The time and size units are now incoherent:

  * **Size** (in bytes) representation:
      * <tt>tr.**b**.units</tt>
          * <tt>SizeInBytes</tt> class which converts numbers to strings (e.g. 1024 → "1.0 KiB"). The class provides both static and non-static formatting methods.
      * <tt>tr.**ui**.units</tt>
          * <tt>\<tr-ui-u-size-in-bytes-span\></tt> element which converts numbers to strings using the <tt>tr.base.units.SizeInBytes.format()</tt> static method.

  * **Time** representation:
      * <tt>tr.**b**.units</tt>
          * <tt>Time</tt> static object which converts numbers to strings depending on the currently selected display unit (e.g. 0.01 → "0.01 ms" or "10,000 ns").
          * <tt>TimeDuration</tt> class, which performs **only basic string conversion** (e.g. 1.2 → "1.200") and has **no tests**.
          * <tt>TimeStamp</tt> class, which performs **only basic string conversion** (e.g. 1.2 → "1.2000"), has **no tests**, and is virtually **identical to <tt>TimeDuration</tt>**.
          * <tt>tsString</tt> utility function, which is **deprecated** and has **no tests**, but is used quite a lot in the rest of the codebase.
      * <tt>tr.**ui**.units</tt>
          * <tt>\<tr-ui-u-time-duration-span\></tt> element which converts numbers to strings using <tt>tr.b.units.Time.currentDisplayUnit.format</tt> (i.e. **completely skipping <tt>tr.base.units.TimeDuration</tt>**)
          * <tt>\<tr-ui-u-time-stamp-span\></tt> element which converts numbers to strings using <tt>tr.b.units.Time.currentDisplayUnit.format</tt> (i.e. **completely skipping <tt>tr.base.units.TimeStamp</tt>**) and is again almost **identical to <tt>\<tr-ui-u-time-duration-span\></tt>**.

To sum up, there are three issues here:
  1. Formatting of <tt>tr.b.units.TimeDuration</tt> and <tt>tr.b.units.TimeStamp</tt> is too basic and is consequently skipped by both <tt>\<tr-ui-u-time-duration-span\></tt> and <tt>\<tr-ui-u-time-stamp-span\></tt>. The two <tt>tr.b.units</tt> classes are also completely untested.
  2. There is a lot of code duplication due to identical behavior of <tt>TimeDuration</tt> (<tt>\<tr-ui-u-time-duration-span\></tt>) and <tt>TimeStamp</tt> (<tt>\<tr-ui-u-time-stamp-span\></tt>).
  3. Many parts of the codebase use the deprecated and untested <tt>tsString</tt> function.

**Comment by natduca** [comment]
Would it make sense for us to go to that units model I suggested in #1057? I wonder if maybe a design doc here is useful.. ~sigh~

**Comment by petrcermak** [comment]
Yeah, I agree that we should compile everything into one design doc to make sure we're consistent. Another thing to consider is the relationship between <tt>tr.model.Attribute</tt> (used by memory dumps) and <tt>tr.b.units.*</tt>.

**Comment by petrcermak** [comment]
This is now mostly fixed (except for some code duplication between time duration and time stamp). Let's move all unit-related discussion to #1057.
